opaque_keys/__init__.py

```diff
@@ -373,8 +373,8 @@ def __ne__(self, other):
         return not self == other
 
     def __lt__(self, other):
-        if type(self) != type(other):
-            raise TypeError('Cannot compare {} to {}.'.format(type(self), type(other)))
+        if (self.KEY_FIELDS, self.CANONICAL_NAMESPACE, self.deprecated) != (other.KEY_FIELDS, other.CANONICAL_NAMESPACE, other.deprecated):
+            raise TypeError("{!r} is incompatible with {!r}".format(self, other))
```

Comment:
Looks OK to me - but which OpaqueKey derivatives are you trying to compare that need this change?